backend/triplestore/rdf_field.py

```diff
@@ -0,0 +1,246 @@
+'''
+Base classes for Django-esque management of RDF data; defines field classes.
+'''
+
+from rdflib.term import Identifier, Literal
+from rdflib import URIRef, Graph
+from typing import Optional
+
+from triplestore import utils
+
+class RDFField:
+    '''
+    Abstract class for RDF fields.
+
+    A subclass of RDFField must implement how the field represents values in a graph and
+    in the Python object. This can be a single triple or a complex structure.
+
+    In most cases, a subclass should implement the following methods:
+    - `__init__`: optional, can be used to set any parameters that apply.
+    - `get`: takes model instance as input and returns the modelled value of the field
+        based on the graph data.
+    - `_stored_triples`: takes a model instance as input and returns the scope of the
+        field; that is, a subset of the graph that the field is modelling.
+    - `_triples_to_store`: returns a collection of triples that represents the modelled
+        value.
+    
+    An `RDFModel` will use the `get`, `set`, and `clear` endpoints. `set` and `clear` have
+    default implementations based on `_stored_triples` and `_triples_to_store`.
+    Alternatively, you could implement `set` and `clear` directly; in that case, it's not
+    necessary to implement `_stored_triples` and `_triples_to_store`.
+
+    The implementations of the methods listed above should use `self.get_graph()` to
+    fetch the correct graph for reading and writing triples. By default, this will be the
+    graph of the model instance. To use a constant graph, you can pass a `graph` in the
+    initialiser. If the graph is chosen dynamically, you can override the method
+    `get_graph`.
+    '''
+
+    def __init__(self, graph: Optional[Graph] = None, **kwargs):
+        self.graph = graph
+
+    def get_graph(self, instance):
```

Comment:
Here and at other places it was not very clear to me immediately what is expected for `instance`. And why is that necessary, if the graph may also be stored in the object during initialization?